core/conversion/converters/impl/plugins/adaptive_max_pool2d_plugin.cpp

```diff
@@ -0,0 +1,317 @@
+    int index,
+    const nvinfer1::DataType* inputTypes,
+    int nbInputs) const {
+  // return DataType::kFLOAT;
+  if (index == 0) {
+    return nvinfer1::DataType::kFLOAT;
+  } else {
+    // return nvinfer1::DataType::kFLOAT;
+    return nvinfer1::DataType::kINT32;
+  }
+}
+
+int AdaptiveMaxPool2dPlugin::initialize() {
+#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
+  tensor_options_ = tensor_options_.device(c10::kCUDA);
+  index_tensor_options_ = index_tensor_options_.device(c10::kCUDA);
+#else
+  tensor_options_ = tensor_options_.device(c10::kCPU);
+  index_tensor_options_ = index_tensor_options_.device(c10::kCPU);
+#endif
+
+  // c10::kFloat = FLOAT32
+  tensor_options_ = tensor_options_.dtype(c10::kFloat);
+  index_tensor_options_ = index_tensor_options_.dtype(c10::kLong);
+
+  return 0;
+}
+
+void AdaptiveMaxPool2dPlugin::serialize(void* buffer) const {
+  std::string data = serializeToString();
+  size_t size = getSerializationSize();
+
+  data.copy((char*)buffer, size);
+}
+
+std::string AdaptiveMaxPool2dPlugin::serializeToString() const {
+  torch::serialize::OutputArchive output_archive;
+
+  output_archive.write("in_shape", torch::IValue(in_shape_));
+  output_archive.write("out_shape", torch::IValue(out_shape_));
+  output_archive.write("size", torch::IValue(size_));
+  output_archive.write("mode", torch::IValue(mode_));
+  output_archive.write("align_corners", torch::IValue(align_corners_));
+
+  std::ostringstream data_str;
+  output_archive.save_to(data_str);
+
+  return data_str.str();
+}
+
+size_t AdaptiveMaxPool2dPlugin::getSerializationSize() const {
+  return serializeToString().size();
+}
+
+bool AdaptiveMaxPool2dPlugin::supportsFormatCombination(
+    int pos,
+    const nvinfer1::PluginTensorDesc* inOut,
+    int nbInputs,
+    int nbOutputs) {
+  TRTORCH_ASSERT(0 <= pos && pos <= 2, "There should be exactly 3 connections to the plugin - 1 input, 2 output");
+  TRTORCH_ASSERT(nbInputs == 1, "Expected a single tensor as input to interpolate plugin");
+  TRTORCH_ASSERT(nbOutputs == 2, "Expected two tensors as output to interpolate plugin");
+
+  const PluginTensorDesc& in = inOut[0];
+
+  if (pos == 0) {
+    return (in.type == nvinfer1::DataType::kFLOAT) && (in.format == nvinfer1::TensorFormat::kLINEAR);
+  }
+
+  // pos == 1, accessing information about output tensor
+  const PluginTensorDesc& out = inOut[1];
+
+  return (in.type == out.type) && (in.format == out.format);
+}
+
+void AdaptiveMaxPool2dPlugin::configurePlugin(
+    const nvinfer1::DynamicPluginTensorDesc* in,
+    int nbInputs,
+    const nvinfer1::DynamicPluginTensorDesc* out,
+    int nbOutputs) {
+  dtype_ = DataType::kFLOAT;
+}
+
+size_t AdaptiveMaxPool2dPlugin::getWorkspaceSize(
+    const nvinfer1::PluginTensorDesc* inputs,
+    int nbInputs,
+    const nvinfer1::PluginTensorDesc* outputs,
+    int nbOutputs) const {
+  return 0;
+}
+
+int AdaptiveMaxPool2dPlugin::enqueue(

```

Comment:
I think that we can encapsulate all aten based plugins in a single plugin with some sort of dispatching mechanism instead of creating classes for each op we want to run in ATen. We should already be doing this for the other interpolate ops.

Reply:
this plugin has two outputs which is different from other single output plugin, there would be two much if else in one single plugin

Reply:
so can we merge this PR, i will seperate PRs in single feature in the future submits